### DISCUSSION TITLE
Programmatic configuration

### DISCUSSION THREAD

**Discussion by andrewzah** [requarks/wiki#4664] [Help / Questions]
Hi, is it possible to programmatically set up and configure wikijs, without user interaction?

Ideally I'd like to be able to skip the setup, and configure LDAP as well.

**Comment by ieaster1** [comment]
Nothing official here, but as far as I can tell, the setup is heavily dependent on an existing DB.  So you could always have a DB and config.yml pair and just set them into place with your automation tools.

**Reply by andrewzah** [reply]
Is there documentation anywhere on what is needed for setting up the db?

**Reply by ieaster1** [reply]
Doc wise, I don't believe so.  Knex.js is used to manage DB migrations though.  You could always view: https://github.com/Requarks/wiki/tree/dev/server/db/migrations

Or, just run through a setup 1-time and copy that data after the initialization.
